packages/descendants/src/DescendantsReducer.ts

```diff
@@ -0,0 +1,138 @@
+import { ComponentProps, Reducer } from 'react';
+import isEqual from 'lodash/isEqual';
+import isUndefined from 'lodash/isUndefined';
+
+import {
+  findDescendantIndexWithId,
+  findDOMIndex,
+  insertDescendantAt,
+  removeIndex,
+} from './utils/';
+import { Descendant, DescendantsList } from './Descendants.types';
+
+export interface DescendantsState<T extends HTMLElement> {
+  descendants: DescendantsList<T>;
+}
+
+export type DescendantsReducerAction<T extends HTMLElement> =
+  | {
+      type: 'register';
```

Comment:
do we need a read action?

Reply:
I don't think so. You can just read off of `state.descendants`

Reply:
I don't think so. You can just read off of `state.descendants`